include/openthread-types.h

```diff
@@ -336,6 +341,30 @@ typedef enum
 } otDeviceRole;
 
 /**
+ * This structure represents the MAC layer counters.
+ */
+typedef struct otMacCounters
+{
+    uint32_t mTx;                  ///< The total number of transmissions.
+    uint32_t mTxDataAcked;         ///< The number of acked data transmissions.
+    uint32_t mTxData;              ///< The number of transmitted data that does not request ack.
+    uint32_t mTxDataPoll;          ///< The number of transmitted data poll.
+    uint32_t mTxBeacon;            ///< The number of transmitted beacon.
+    uint32_t mTxBeaconRequest;     ///< The number of transmitted beacon request.
+    uint32_t mTxDataUnAcked;       ///< The number of unacked data transmissions.
```

Comment:
The acked vs. unacked is still a bit confusing to me.  What we need with respect to that:

- `mTxAckRequested`: the number of transmissions with ack request
- `mTxAcked:` the number of transmissions that were successfully acked
- `mTxNoAckRequested:` the number of transmissions without ack request

Note that `mTxAckRequested` + `mTxNoAckRequested` should equal `mTxTotal`.

Also, I think that `mTxAckRequested` - `mTxAcked` should equal `mTxRetry`

Then we have a separate category for different MAC frame types:
- `mTxData`
- `mTxDataPoll`
- `mTxBeacon`
- `mTxBeaconRequest`

Note that for Thread, the sum of above should also equal `mTxTotal`.